src/Npgsql/ConnectorPool.cs

```diff
@@ -319,6 +324,8 @@ internal override void Clear()
                 count--;
             }
         }
+
+        _isClearing = 0;
```

Comment:
Put in finally?

Reply:
Was thinking about that: the only potential place which can throw an exception is `CloseConnector`, and even then it should only happen if we can't find the slot the connector had. But sure.

Reply:
Sure - this is only for extra safety (and in case some new code gets added in the future which can throw)